torch/csrc/autograd/function.h

```diff
@@ -119,11 +119,9 @@ struct TORCH_API Node : std::enable_shared_from_this<Node> {
       assign_parent();
     }
 
-    if (profiler::profilerEnabled()) {
-      // If profiler is enabled, thread_id is stored.
-      // See NOTE [ Sequence Numbers ]
-      thread_id_ = at::RecordFunction::currentThreadId();
-    }
+    // Store the thread_id of the forward operator.
+    // See NOTE [ Sequence Numbers ]
+    thread_id_ = at::RecordFunction::currentThreadId();
```

Comment:
Why do you need to change that?
What is the perf cost of always doing this lookup?

Reply:
Hi @albanD, `profilerEnabled()` and profiler_legacy is being deprecated (please see this [conversation](https://www.internalfb.com/diff/D27238906?dst_version_fbid=132461845501449&transaction_fbid=445031943390803)). `currentThreadId()` returns the an uint64_t, and is set once per thread. The cost should be almost the same as we are removing a conditional and `currentThreadId()` internally does a check on the uint64_t has been set and returns the value.

Reply:
I understand you want to remove that function, but we should still make sure there isn't any perf issue related to this.
You can use: https://pytorch.org/docs/master/benchmark_utils.html?highlight=timer#torch.utils.benchmark.Timer in instruction count mode to easily benchmark this.

Reply:
Sure, I could run a comparison for before/after. Do you have any recommendation on which benchmarks use for this?

Reply:
Any code that operates on Tensors that require gradients will end up here, so something like this should work:
```
a = torch.rand(1, requires_grad=True)

b = a + 1
```

Reply:
Thanks, I just copied the examples from the benchmark tutorial page and added `requires_grad=True`. 

Source code and instruction count comparison:

P404935604

To summarize:
```
### BEFORE
                           All          Noisy symbols removed
    Instructions:      3107840                    3085414
    Baseline:             4252                       3844

### AFTER
                           All          Noisy symbols removed
    Instructions:      3101781                    3075609
    Baseline:             4236                       3844
```
So it looks like we get some savings in terms of instructions.

Reply:
Awesome, thanks for checking!